packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -1373,10 +1366,23 @@ module.exports = function<T, P, I, TI, PI, C, CC, CX, PL>(
     let previousScheduledRoot = null;
     let root = firstScheduledRoot;
     while (root !== null) {
+      if (isUnbatchingUpdates) {
+        if (root.isUnbatched) {
```

Comment:
Hm. I wonder if we can just pass the root specifically to "flush" since we know which one we're invoking render/unmount on. What do you think?